includes/compat/class-wc-stripe-subscriptions-helper.php

```diff
@@ -103,4 +141,59 @@ public static function is_manual_renewal_enabled() {
 		}
 		return false;
 	}
+
+	/**
+	 * Builds a string containing messages about subscriptions that are detached from the customer.
+	 *
+	 * @param $subscriptions array An array of subscriptions that are detached from the customer.
+	 * @return string A string containing the messages to be displayed in the admin interface.
+	 */
+	public static function build_subscriptions_detached_messages( $subscriptions = [] ) {
+		if ( empty( $subscriptions ) || ! is_array( $subscriptions ) ) {
+			return '';
+		}
+
+		$detached_messages = '';
+		foreach ( $subscriptions as $subscription ) {
+			$customer_payment_method_link = sprintf(
+				'<a href="%s">%s</a>',
+				esc_url( $subscription['change_payment_method_url'] ),
+				esc_html(
+				/* translators: this is a text for a link pointing to the customer's payment method page */
+					__( 'Payment method page &rarr;', 'woocommerce-gateway-stripe' )
+				)
+			);
+			$customer_stripe_page = sprintf(
+				'<a href="%s">%s</a>',
+				esc_url( self::STRIPE_CUSTOMER_PAGE_BASE_URL . $subscription['customer_id'] ),
+				esc_html(
+				/* translators: this is a text for a link pointing to the customer's page on Stripe */
+					__( 'Stripe customer page &rarr;', 'woocommerce-gateway-stripe' )
+				)
+			);
+			$detached_messages .= sprintf(
+			/* translators: %1$s is the subscription ID. %2$s is a customer payment method page. %3$s is the customer's page on Stripe */
+				__( '#%1$s: %2$s | %3$s<br/>', 'woocommerce-gateway-stripe' ),
+				$subscription['id'],
```

Comment:
I think we should wrap this value in `esc_html()` just to be safe.
```suggestion
				esc_html( $subscription['id'] ),
```